tests/telemetry/test_telemetry_metrics_e2e.py

```diff
@@ -172,7 +172,7 @@ def test_span_creation_and_finished_metrics_opentracing(test_agent_session, ddtr
 def test_span_creation_no_finish(test_agent_session, ddtrace_run_python_code_in_subprocess):
     code = """
 import ddtrace
-import opentelemetry
+import opentelemetry.trace
```

Comment:
I still don't follow why this import is problematic. If we update `@ModuleWatchdog.after_module_imported("opentelemetry.trace")` to `@ModuleWatchdog.after_module_imported("opentelemetry")` would we still need this refactor?

Reply:
`opentelemetry` is a namespace package. When imported, it essentially contains nothing
```
>>> import opentelemetry
>>> dir(opentelemetry)
['__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']
```
The `trace` sub-module needs to be imported directly for it to be added to `opentelemetry`
```
>>> import opentelemetry.trace
>>> dir(opentelemetry)
['__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__', 'attributes', 'context', 'environment_variables', 'trace', 'util']
```
We could in theory make the suggested change `@ModuleWatchdog.after_module_imported("opentelemetry.trace") -> @ModuleWatchdog.after_module_imported("opentelemetry")` (which would require #7727), but because `opentelemetry` is just a namespace package, it is unlikely that somebody would import just that, as it is pretty much useless. The real module of interest is `opentelemetry.trace` here.